src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs

```diff
@@ -78,18 +78,67 @@ internal class Descriptors
             "Internal visible to product libraries effectively become public API and have to be versioned appropriately", "Usage", DiagnosticSeverity.Warning, true);
             
 
-        public static DiagnosticDescriptor AZC0012 = new DiagnosticDescriptor(
-            nameof(AZC0012),
+        public static DiagnosticDescriptor AZC0100 = new DiagnosticDescriptor(
+            nameof(AZC0100),
             "ConfigureAwait(false) must be used.",
             "ConfigureAwait(false) must be used.",
             "Usage",
             DiagnosticSeverity.Warning, true);
 
-        public static DiagnosticDescriptor AZC0013 = new DiagnosticDescriptor(
-            nameof(AZC0013),
+        public static DiagnosticDescriptor AZC0101 = new DiagnosticDescriptor(
+            nameof(AZC0101),
             "Use ConfigureAwait(false) instead of ConfigureAwait(true).",
             "Use ConfigureAwait(false) instead of ConfigureAwait(true).",
             "Usage",
             DiagnosticSeverity.Warning, true);
+
+        public static DiagnosticDescriptor AZC0102 = new DiagnosticDescriptor(
+            nameof(AZC0102),
+            "Do not use GetAwaiter().GetResult().",
+            "Do not use GetAwaiter().GetResult(). Use TaskExtensions.EnsureCompleted() extension method instead.",
+            "Usage",
+            DiagnosticSeverity.Warning, true);
+
+        public static DiagnosticDescriptor AZC0103 = new DiagnosticDescriptor(
+            nameof(AZC0103),
+            "Do not wait synchronously in asynchronous scope.",
+            "Do not use {0} in asynchronous scope. Use await keyword instead.",
+            "Usage",
+            DiagnosticSeverity.Warning, true);
+
+        public static DiagnosticDescriptor AZC0104 = new DiagnosticDescriptor(
+            nameof(AZC0104),
+            "Use EnsureCompleted() directly on asynchronous method return value.",
+            "Don't use EnsureCompleted on the {0}. Call EnsureCompleted() extension method directly on the return value of the asynchronous method that has 'bool async' parameter.",
+            "Usage",
+            DiagnosticSeverity.Warning, true);
+
+        public static DiagnosticDescriptor AZC0105 = new DiagnosticDescriptor(
+            nameof(AZC0105),
+            "DO NOT add 'async' parameter to public methods.",
+            "DO provide both asynchronous and synchronous variants for all service methods instead of one variant with 'async' parameter.",
```

Comment:
```suggestion
            "DO provide both synchronous and asynchronous variants for all service methods instead of one variant with the 'async' parameter.",
```